aviary/subsystems/propulsion/propeller_map.py

```diff
@@ -0,0 +1,144 @@
+import numpy as np
+import warnings
+import openmdao.api as om
+from openmdao.utils.units import convert_units
+
+from aviary.subsystems.propulsion.utils import PropModelVariables, default_prop_units
+from aviary.utils.aviary_values import AviaryValues, NamedValues, get_keys
+from aviary.utils.csv_data_file import read_data_file
+from aviary.utils.functions import get_path
+from aviary.variable_info.enums import OutMachType
+from aviary.variable_info.variables import Aircraft, Settings
+
+
+MACH = PropModelVariables.MACH
+CP = PropModelVariables.CP
+CT = PropModelVariables.CT
+J = PropModelVariables.J
+
+aliases = {
+    # whitespaces are replaced with underscores converted to lowercase before
+    # comparison with keys
+    MACH: ['m', 'mn', 'mach', 'mach_number'],
+    CP: ['cp', 'power_coefficient'],
+    CT: ['ct', 'thrust_coefficient'],
+    J: ['j', 'advance_ratio'],
+}
+
+
+class PropellerMap(om.ExplicitComponent):
+    """
+    This class loads a user provided propeller map into memory and builds a propeller.
+    Attributes
+    ----------
+    name : str ('propeller')
+        Object label.
+    options : AviaryValues (<empty>)
+        Inputs and options related to propeller model.
+    data : NamedVaues (<empty>), optional
+        propeller map data.
+    mach_type: OutMachType (MACH or HELICAL_MACH)
+    """
+
+    def __init__(self, name='propeller', options: AviaryValues = None,
+                 data: NamedValues = None, mach_type=OutMachType.MACH):
```

Comment:
Mach type is something that is set by the data and can't be changed - having Mach type as an argument allows the user to make mistakes in how Mach number is defined. A possible solution is to have a different data header for helical Mach so this component can differentiate.

That could be a follow-on after this PR (I can make an issue to address later)

Reply:
I removed `mach_type` as an argument.